api/src/main/java/marquez/api/filter/CorsFilter.java

```diff
@@ -97,4 +119,13 @@ protected void preflight(String origin, ContainerRequestContext requestContext)
         requestContext.abortWith(builder.build());
 
     }
+
+    protected void checkOrigin(ContainerRequestContext requestContext, String origin) {
+        if (!getAllowedOrigins().contains("*") && !CorsUtils.tryMatchAny(origin, new ArrayList<>(getAllowedOrigins()))) {
+            requestContext.setProperty("cors.failure", true);
+            throw new ForbiddenException(String.format("Origin not allowed: %s", origin));
```

Comment:
nit: `String#format` calls are obscenely expensive, since we're just concatenating here I'd probably rewrite this as
```suggestion
            throw new ForbiddenException("Origin not allowed: " + origin);
```